diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -394,8 +394,16 @@ class Updater : public DiagnosticTaskVector
     node_name_(base_interface->get_name()),
     warn_nohwid_done_(false)
   {
-    period = parameters_interface->declare_parameter(
-      "diagnostic_updater.period", rclcpp::ParameterValue(period)).get<double>();
+    constexpr const char * period_param_name = "diagnostic_updater.period";
+    rclcpp::ParameterValue period_param;
+    try {
+      period_param = parameters_interface->declare_parameter(
+        period_param_name, rclcpp::ParameterValue(
+          period));
+    } catch (const rclcpp::exceptions::ParameterAlreadyDeclaredException & exc) {
+      period_param = parameters_interface->get_parameter(period_param_name).get_parameter_value();
+    }
```

Comment:
```suggestion
    if (parameters_interface->has_parameter(period_param_name)) {
      period_param = parameters_interface->get_parameter(period_param_name).get_parameter_value();
    } else {
      period_param = parameters_interface->declare_parameter(
        period_param_name, rclcpp::ParameterValue(period));
    }
```

@bergercookie, what do you think of this alternative implementation without try-catch, proposed by one of my colleagues? This is in line with @com2rng's previous comment.

Reply:
I have to test whether this works. Looks like a reasonable suggestion.

Can someone from the repo admins have a look and review as well?
This PR has been sitting here for ~6months and I don't want to put more effort if it's not going to be merged :)

Reply:
I think @ralph-lange is allowed to merge this PR and we are actually more than welcome that you already worked on this!

Reply:
Thanks @ralph-lange for merging it :-)

Reply:
Kindly excuse that it took so long. I should have clarified the maintenance issue earlier. I'll make new bloom releases for all relevant ROS 2 distributions as soon as I have access to https://github.com/ros2-gbp/diagnostics-release.